### PR TITLE
Delete deprecated method Image::storePhoto()

### PR DESCRIPTION
**Pull request by nupplaphil**
<strike>During replacing `q` calls, I found this static method defined in a very wrong place of the codebase.
=> So I moved it out of `Object` into `Model`.

I saw some improvements in the code too, but I don't want to shake the boat too much.</strike>

Deleted the whole method `Image::storePhoto()` because of no usage left.
Reason: https://github.com/friendica/friendica-addons/pull/835

### CODE REVIEW

**Comment by nupplaphil** [comment]
Has to get merged with https://github.com/friendica/friendica-addons/pull/818

File: src/Model/Photo.php [hunk 1]
```diff
@@ -351,6 +351,178 @@ public static function delete(array $conditions, array $options = [])
...
+		$height = $Image->getHeight();
+
+		$hash = Photo::newResource();
+
+		// Pictures are always public by now
+		//$defperm = '<'.$default_cid.'>';
+		$defperm = "";
+		$visitor = 0;
+
+		$r = Photo::store($Image, $uid, $visitor, $hash, $tempfile, L10n::t('Wall Photos'), 0, 0, $defperm);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Shouldn't it be `self::store`?

File: src/Model/Photo.php [hunk 2]
```diff
@@ -351,6 +351,178 @@ public static function delete(array $conditions, array $options = [])
...
+
+		// Unsure, if this should be activated or deactivated
+		//if (isset($image["small"]))
+		//  $image["preview"] = $image["small"];
+
+		if (isset($image["medium"])) {
+			$image["preview"] = $image["medium"];
+		}
+
+		return($image);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Code standards: Please remove the parentheses around `return` calls.

File: src/Model/Photo.php [hunk 3]
```diff
@@ -351,6 +351,178 @@ public static function delete(array $conditions, array $options = [])
...
+		$image = ["page" => System::baseUrl().'/photos/'.$page_owner_nick.'/image/'.$hash,
+			"full" => $a->getBaseURL()."/photo/{$hash}-0.".$Image->getExt()];
+
+		if ($width > 800 || $height > 800) {
+			$image["large"] = System::baseUrl()."/photo/{$hash}-0.".$Image->getExt();
+		}
+
+		if ($width > 640 || $height > 640) {
+			$Image->scaleDown(640);
+			$r = Photo::store($Image, $uid, $visitor, $hash, $tempfile, L10n::t('Wall Photos'), 1, 0, $defperm);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Shouldn't it be `self::store`?

File: src/Model/Photo.php [hunk 4]
```diff
@@ -351,6 +351,178 @@ public static function delete(array $conditions, array $options = [])
...
+			if ($r) {
+				$image["medium"] = System::baseUrl()."/photo/{$hash}-1.".$Image->getExt();
+			}
+		}
+
+		if ($width > 320 || $height > 320) {
+			$Image->scaleDown(320);
+			$r = Photo::store($Image, $uid, $visitor, $hash, $tempfile, L10n::t('Wall Photos'), 2, 0, $defperm);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Shouldn't it be `self::store`?

File: src/Model/Photo.php [hunk 5]
```diff
@@ -369,28 +369,28 @@ public static function storeByData($uid, $imagedata = "", $url = "")
 		$logger = $a->getLogger();
 		$profiler = $a->getProfiler();
 
-		$r = DBA::p(
+		$isStored = DBA::p(
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Please replace this call by `User::getOwnerById`

File: src/Model/Photo.php [hunk 6]
```diff
@@ -399,109 +399,109 @@ public static function storeByData($uid, $imagedata = "", $url = "")
...
 		}
 
-		$maximagesize = Config::get('system', 'maximagesize');
+		$maxImageSize = Config::get('system', 'maximagesize');
 
-		if (($maximagesize) && (strlen($imagedata) > $maximagesize)) {
-			$logger->info("Image exceeds size limit.", ['max' => $maximagesize, 'current' => strlen($imagedata)]);
-			return([]);
+		if (($maxImageSize) && (strlen($imagedata) > $maxImageSize)) {
+			$logger->info("image exceeds size limit.", ['max' => $maxImageSize, 'current' => strlen($imagedata)]);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Please leave the capital in `Image` in logger message.

File: src/Model/Photo.php [hunk 7]
```diff
@@ -399,109 +399,109 @@ public static function storeByData($uid, $imagedata = "", $url = "")
...
 		if (!isset($data["mime"])) {
-			unlink($tempfile);
+			unlink($tempFile);
 			$logger->info("File is no picture");
-			return([]);
+			return [];
 		}
 
-		$Image = new Image($imagedata, $data["mime"]);
+		$image = new Image($imagedata, $data["mime"]);
```
Comment (MrPetovan) [inline src/Model/Photo.php]: I like have variable containing object start with a capital.

File: src/Model/Photo.php (hunk 5 above)
Comment (MrPetovan) [inline src/Model/Photo.php]: Are you sure about `$isStored`?

File: src/Model/Photo.php (hunk 7 above)
Comment (nupplaphil) [reply src/Model/Photo.php]: Hm.. What about `$objImage` ? I would prefers not to name any local variable with a started capitalcapital

Comment (MrPetovan) [reply src/Model/Photo.php]: I'm not a fan of the Hungarian notation, and the starting capital for object variables may just a personal preference that doesn't need to persist.

File: src/Model/Photo.php [hunk 8]
```diff
@@ -369,20 +369,20 @@ public static function storeByData($uid, $imagedata = "", $url = "")
 		$logger = $a->getLogger();
 		$profiler = $a->getProfiler();
 
-		$isStored = DBA::p(
+		$stmtUser = DBA::p(
```
Comment (MrPetovan) [inline src/Model/Photo.php]: I'd prefer `$userStmt` if you don't mind.

File: src/Model/Photo.php [hunk 9]
```diff
@@ -351,6 +351,178 @@ public static function delete(array $conditions, array $options = [])
 		return DBA::delete("photo", $conditions, $options);
 	}
 
+	/**
+	 * @brief This function is used by the fromgplus addon
```
Comment (annando) [inline src/Model/Photo.php]: You know that the addon will be removed in that version since G+ isn't anymore?

File: src/Model/Photo.php (hunk 5 above)
Comment (nupplaphil) [reply src/Model/Photo.php]: What about the additional `user.blocked = 0` ? This wouldn't be included

**Comment by nupplaphil** [comment]
It's getting the way I didn't want to go. I found this implementation at a place, where it doesn't belong to. But now I'm starting refactoring the code itself ;-)

@annando will the g+ addon get removed or just deprecated? If one, we can simply delete this method too and I don't take my time to fix it up :-)

**Comment by MrPetovan** [comment]
> It's getting the way I didn't want to go.

What do you mean?

**Comment by nupplaphil** [comment]
Sorry, I hope this didn't sound offensive.

I added the PR with the comment:
> I saw some improvements in the code too, but I don't want to shake the boat too much.

Because I know that there are some improvements, but I'm not able to fully test this part of code, so I thought "just" moving it to another place wouldn't create noise. Now I'm no more sure because I'm replacing parts of the logic. And it seems that this code will get deprecated/removed with the deletion of the g+ addon.

**Comment by MrPetovan** [comment]
Ah, I see what you mean now, and you weren't offensive, thanks for the elaboration. Believe me, in 2 years working on Friendica, I know the feeling!

**Comment by MrPetovan** [comment]
This needs a rebase.

**Comment by nupplaphil** [comment]
https://github.com/friendica/friendica/pull/6784#discussion_r261769629
Does it really work? because of the "blocked" parameter. Shall I extend `User::getOwnerById()` ?

**Comment by MrPetovan** [comment]
No, you can just check the value of `$owner['blocked']` after calling `User::getOwnerById()`.

File: src/Model/Photo.php (hunk 9 above)
Comment (MrPetovan) [reply src/Model/Photo.php]: The addon has been removed, please check if this method is used anywhere else.

**Comment by nupplaphil** [comment]
Anything left? :-) I do want to close at least one of my long time PRs hehe

**Comment by MrPetovan** [comment]
I was waiting for the tests and as usual I ended up forgetting about the PR.

**Comment by nupplaphil** [comment]
"no usages found". Shall I delete the whole method and rename this PR? ;-)

**Pull request merged**